sql/migrate/dir.go

```diff
@@ -555,3 +556,79 @@ func (m *memFile) Mode() fs.FileMode          { return 0 }
 func (m *memFile) ModTime() time.Time         { return time.Time{} }
 func (m *memFile) IsDir() bool                { return false }
 func (m *memFile) Sys() interface{}           { return nil }
+
+// ArchiveDir returns a tar archive of the given directory.
+func ArchiveDir(dir Dir) ([]byte, error) {
+	var buf bytes.Buffer
```

Comment:
I think we should validate the dir here once. Atlas promises to not work on broken files.

Reply:
but we generate the sumfile ourselves in line 566

Reply:
You can try to call Open(HashFileName), and check exist error.

Reply:
@masseelch: I just thought, maybe we need to accept an invalid dir and then later report it using atlas lint?

Reply:
I don’t think it’s the responsibility of this function to validate the Dir. However, it does make sense to call migrate.Validate in Unarchive as we build and return a Dir, and I expect to get a valid one.